crates/ostree-ext/src/chunking.rs

```diff
@@ -1221,4 +1230,183 @@ mod test {
 
         Ok(())
     }
+
+    #[test]
+    fn test_process_mapping_specific_components_contain_correct_objects() -> Result<()> {
+        // This test validates that specific components get their own dedicated layers
+        // and that their objects are properly isolated from regular package layers
+
+        // Setup: Create 5 packages
+        // - pkg1, pkg2: Will be marked as "specific components" (get their own layers)
+        // - pkg3, pkg4, pkg5: Regular packages (will be bin-packed together)
+        let packages = [
+            (1, 100, 50000), // pkg1 - SPECIFIC COMPONENT
+            (2, 200, 40000), // pkg2 - SPECIFIC COMPONENT
+            (3, 300, 30000), // pkg3 - regular package
+            (4, 400, 20000), // pkg4 - regular package
+            (5, 500, 10000), // pkg5 - regular package
+        ];
+
+        let (contentmeta, mut system_metadata, mut specific_components_meta, mut chunking) =
+            setup_exclusive_test(&packages, 8, None)?;
+
+        // Create object mappings
+        // - system_objects_map: Contains ALL objects in the system (both specific and regular)
+        // - specific_components_objects_map: Contains ONLY objects from specific components
+        let mut system_objects_map = IndexMap::new();
+        let mut specific_components_objects_map = IndexMap::new();
+
+        // SPECIFIC COMPONENT 1 (pkg1): owns 3 objects
+        let pkg1_objects = ["checksum_1_a", "checksum_1_b", "checksum_1_c"];
```

Comment:
OK as is but in theory we should actually require these to be at least string SHA256 digests.